plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/controls/CustomCheckboxCellEditor.java

```diff
@@ -52,8 +52,16 @@ public class CustomCheckboxCellEditor extends CellEditor implements BooleanStyle
     private UIElementAlignment alignment;
 
     public CustomCheckboxCellEditor(Composite parent) {
+        this(parent, false);
+    }
+    public CustomCheckboxCellEditor(Composite parent, boolean changeOnActivate) {
         super(parent);
-
+        this.changeOnActivate = changeOnActivate;
+        //added to prevent checkbox flicking when changeOnActivate is true
+        if (!changeOnActivate) {
+            checkBox.setBackground(parent.getBackground());
```

Comment:
```suggestion
        if (!changeOnActivate) {
            // added to prevent checkbox flickering when the changeOnActivate is set
            checkBox.setBackground(parent.getBackground());
```